compiler/packages/babel-plugin-react-compiler/src/HIR/Environment.ts

```diff
@@ -242,9 +242,41 @@ const EnvironmentConfigSchema = z.object({
   enableOptionalDependencies: z.boolean().default(true),
 
   /**
-   * Enables inference and auto-insertion of effect dependencies. Still experimental.
+   * Enables inference and auto-insertion of effect dependencies. Takes in an array of
+   * configurable module and import pairs to allow for user-land experimentation. For example,
+   * [
+   *   {
+   *     module: 'react',
+   *     imported: 'useEffect',
+   *     numRequiredArgs: 1,
+   *   },{
+   *     module: 'MyExperimentalEffectHooks',
+   *     imported: 'useExperimentalEffect',
+   *     numRequiredArgs: 2,
+   *   },
+   * ]
+   * would insert dependencies for calls of `useEffect` imported from `react` and calls of
+   * useExperimentalEffect` from `MyExperimentalEffectHooks`.
+   *
+   * `numRequiredArgs` tells the compiler the amount of arguments required to append a dependency
+   *  array to the end of the call. With the configuration above, we'd insert dependencies for
+   *  `useEffect` if it is only given a single argument and it would be appended to the argument list.
+   *
+   * numRequiredArgs must always be greater than 0, otherwise there is no function to analyze for dependencies
+   *
+   * Still experimental.
    */
-  inferEffectDependencies: z.boolean().default(false),
+  inferEffectDependencies: z
+    .nullable(
+      z.array(
+        z.object({
+          module: z.string(),
+          imported: z.string(),
+          numRequiredArgs: z.number(),
+        }),
+      ),
+    )
```

Comment:
tiny nit (feel free to ignore), it would be nice to make this consistent with other external function config options by using the [`ExternalFunctionSchema`](https://github.com/facebook/react/blob/6f0dc2947bed21f9be484f37eb32d02fdc4c0481/compiler/packages/babel-plugin-react-compiler/src/HIR/Environment.ts#L61C1-L68C1) type.

Reply:
ohhh nice!